### PR TITLE
prepare codebase to remove array polyfills and trimmed down babel-helpers file

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
   "parser": "babel-eslint",
   "ecmaFeatures": {
+    "modules": true,
     "arrowFunctions": true,
     "blockBindings": true,
-    "forOf": false
+    "forOf": false,
+    "destructuring": false,
```
Comment (cramforce) [inline .eslintrc]: Could we turn this on? Impact should be minimal, right?

File: build-system/tasks/presubmit-checks.js [hunk 2]
```diff
@@ -54,6 +54,14 @@ var forbiddenTerms = {
   'Promise\\.race': es6polyfill,
   '\\.startsWith': es6polyfill,
   '\\.endsWith': es6polyfill,
+  // TODO: (erwinm) rewrite the destructure and spread warnings as
+  // eslint rules (takes more time than this quick regex fix).
+  // No destructuring allowed since we dont ship with Array polyfills.
```
Comment (cramforce) [inline build-system/tasks/presubmit-checks.js]: Ignore above then :)

**Comment by cramforce** [comment]
What is the code size impact?

**Comment by erwinmombay** [comment]
@cramforce here are the new sizes

**Comment by erwinmombay** [comment]
@cramforce here are the new sizes https://github.com/ampproject/amphtml/pull/721/files

big difference is in integration.js and f.js

screen shot 2015-10-22 at 8 36 34 am

we could allow for destructuring in core just not in 3p and ads (not sure but might be worth looking into)

**Comment by dvoytenko** [comment]
LGTM from me

**Comment by cramforce** [comment]
LGTM
Nice

**Pull request merged**